compiler/passes/src/type_checking/check_program.rs

```diff
@@ -70,16 +71,22 @@ impl<'a> ProgramVisitor<'a> for TypeChecker<'a> {
         }
 
         for Member { identifier, type_ } in input.members.iter() {
-            // Ensure there are no tuple typed members.
-            self.assert_not_tuple(identifier.span, type_);
+            // Check that the member type is not a tuple.
+            if matches!(type_, Type::Tuple(_)) {
```

Comment:
I think this probably answers my question in the comment above. We prohibit that in the definition of struct and record types, and so a struct or record expression with a tuple as component would just not type-check.